File: push-notification-backend/src/main/java/com/hathatinggroot/pushnotification/domain/Notification.java
package com.hathatinggroot.pushnotification.domain;


import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
public class Notification {
    private String id;
    private long timestamp;
    private String senderId;
    private String topic;
    private String content;

    private String receiverId;
    private boolean read;

    public Notification() {
        this.id = UUID.randomUUID().toString();
        this.timestamp = System.currentTimeMillis();
    }

    public Notification(String senderId, String topic, String content) {
        this();
        this.senderId = senderId;
        this.topic = topic;
        this.content = content;
    }
}

File: frontend/src/domain/Notification.ts
export default class Notification {
    id: string;
    timestamp: number;
    senderId: string;
    topic: string;
    content: string;

    receiverId: string;
    read: boolean;

    constructor(id: string, timestamp: number, senderId: string, topic: string, content: string, receiverId: string, read: boolean) {
        this.id = id;
        this.timestamp = timestamp;
        this.senderId = senderId;
        this.topic = topic;
        this.content = content;
        this.receiverId = receiverId;
        this.read = read;
    }

    static fromBody(body: any) {
        console.log('from body', body);
        return new Notification(
            body.id,
            body.timestamp,
            body.senderId,
            body.topic,
            body.content,
            body.receiverId,
            body.read
        )
    }
}